akka-stream-typed/src/main/scala/akka/stream/typed/javadsl/ActorSource.scala

```diff
@@ -47,6 +48,7 @@ object ActorSource {
    * @param bufferSize The size of the buffer in element count
    * @param overflowStrategy Strategy that is used when incoming elements cannot fit inside the buffer
    */
+  @deprecated("Use `actorRef` that takes japi function instead", "2.6.0")
```

Comment:
Since this PR was backported to 2.5.23 (with this deprecation warning) and we have ApiMayChange here I don't think we need a long deprecation cycle for this. Let's remove this method in master (2.6.0). Short mention the replacement in `migration-guide-2.5.x-2.6.x.md`.

Reply:
Ok, will update the PR